source/CMakeLists.txt

```diff
@@ -19,19 +29,24 @@ set(SRCS
 
 find_package (Threads REQUIRED) #threading
 
-#-------------------------------------------------------------------------------
-#exampleApp executable setup
-if(WIN32) #windows icon resource file, provided from /packaging directory
-    set(WINDOWS_ICON_RESOURCE "${PROJECT_BINARY_DIR}/exampleApp.icon.rc")
-endif()
-
-add_executable(exampleApp ${SRCS} ${WINDOWS_ICON_RESOURCE})
+add_executable(exampleApp MACOSX_BUNDLE ${SRCS} ${ICON_RESOURCE})
 target_link_libraries (exampleApp ${CMAKE_THREAD_LIBS_INIT}) #threading
+
 if(NOT MSVC)
     target_compile_options(exampleApp PRIVATE -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wunused -pedantic) # extra warnings
 endif()
+
+if(APPLE)
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "exampleApp")
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_BUNDLE_GUI_IDENTIFIER "com.exampleApp.exampleApp")
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_ICON_FILE exampleApp.icns)
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${PROJECT_SOURCE_DIR}/packaging/MacOSXBundleInfo.plist.in)
+    set_source_files_properties(${PROJECT_SOURCE_DIR}/packaging/exampleApp.icns PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
+endif()
+
 set_property(TARGET exampleApp PROPERTY CXX_STANDARD 14) # we want C++14
-install(TARGETS exampleApp DESTINATION bin) #copy executable to bin folder on install
+# Bundle destination specific for Mac DMG.
+install(TARGETS exampleApp BUNDLE DESTINATION "." RUNTIME DESTINATION bin)
```

Comment:
Just to be sure: So for apple it is necessary for binaries to be in root folder of package? 

If this has to be this way, then I will go ahead and merge this, But if binaries can be in bin folder in packages, I think it would be preferable that they are in bin folder. If that is not possible, then no problem. 

Thanks for all your work. :)